src/main/java/frc/robot/IMU/BNO055.java

```diff
@@ -0,0 +1,776 @@
+		   2 = System initialization error
+		   3 = Self test result failed
+		   4 = Register map value out of range
+		   5 = Register map address out of range
+		   6 = Register map write error
+		   7 = BNO low power mode not available for selected operation mode
+		   8 = Accelerometer power mode not available
+		   9 = Fusion algorithm configuration error
+		   A = Sensor configuration error */
+		status.system_error = read8(reg_t.BNO055_SYS_ERR_ADDR);
+		return status;
+	}
+
+	/**
+	 * Gets the chip revision numbers
+	 *
+	 * @return the chips revision information
+	 */
+	public RevInfo getRevInfo() {
+		int a = 0, b = 0;
+		RevInfo info = new RevInfo();
+
+		/* Check the accelerometer revision */
+		info.accel_rev = read8(reg_t.BNO055_ACCEL_REV_ID_ADDR);
+
+		/* Check the magnetometer revision */
+		info.mag_rev   = read8(reg_t.BNO055_MAG_REV_ID_ADDR);
+
+		/* Check the gyroscope revision */
+		info.gyro_rev  = read8(reg_t.BNO055_GYRO_REV_ID_ADDR);
+
+		/* Check the SW revision */
+		info.bl_rev    = read8(reg_t.BNO055_BL_REV_ID_ADDR);
+
+		a = read8(reg_t.BNO055_SW_REV_ID_LSB_ADDR);
+		b = read8(reg_t.BNO055_SW_REV_ID_MSB_ADDR);
+		info.sw_rev = (short) ((b << 8) | a);
+
+		return info;
+	}
+
+	/**
+	 * Diagnostic method to determine if communications with the sensor are active.
+	 *   Note this method returns true after first establishing communications
+	 *   with the sensor.
+	 *   Communications are not actively monitored once sensor initialization
+	 *     has started.
+	 * @return true if the sensor is found on the I2C bus
+	 */
+	public boolean isSensorPresent() {
+		return sensorPresent;
+	}
+
+	/**
+	 * After power is applied, the sensor needs to be configured for use.
+	 *   During this initialization period the sensor will not return position
+	 *   vector data. Once initialization is complete, data can be read,
+	 *   although the sensor may not have completed calibration.
+	 *   See isCalibrated. 
+	 * @return true when the sensor is initialized.
+	 */
+	public boolean isInitialized() {
+		return initialized;
+	}
+	
+	/**
+	 * Gets current calibration state.
+	 * @return each value will be set to 0 if not calibrated, 3 if fully
+	 *   calibrated.
+	 */
+	public CalData getCalibration() {
+		CalData data = new CalData();
+		int rawCalData = read8(reg_t.BNO055_CALIB_STAT_ADDR);
+
+		data.sys = (byte) ((rawCalData >> 6) & 0x03);
+		data.gyro = (byte) ((rawCalData >> 4) & 0x03);
+		data.accel = (byte) ((rawCalData >> 2) & 0x03);
+		data.mag = (byte) (rawCalData & 0x03);
+
+		return data;
+	}
+
+	/**
+	 * Returns true if all required sensors (accelerometer, magnetometer,
+	 *   gyroscope) have completed their respective calibration sequence.
+	 *   Only sensors required by the current operating mode are checked.
+	 *   See Section 3.3.
+	 * @return true if calibration is complete for all sensors required for the
+	 *   mode the sensor is currently operating in. 
+	 */
+	public boolean isCalibrated() {
```

Comment:
Something like this for just accelerometer and gyroscope would be good.  We don't really care about the magnetometer right now.